javascript/javascript3/week1/weather_app/index.js

```diff
@@ -0,0 +1,101 @@
+//@ts-check
+
+window.addEventListener('load', () => {
+    // Load the last searched city from the local storage
+    const cityFromLocalStorage = localStorage.getItem('city');
+    console.log(cityFromLocalStorage);
+    if (cityFromLocalStorage) {
+        getWeather(`https://api.openweathermap.org/data/2.5/weather?q=${cityFromLocalStorage}&appid=0b2171133666adf69c5fc412fc18bb00`);
+    }
+});
+
+const cityInputField = document.querySelector('#city_input_field');
+const selectButton = document.querySelector('#select_button');
+const weatherInfo = document.querySelector('#weather_info');
+const city = document.querySelector('#city');
+const temperature = document.querySelector('#temperature');
+const weatherIcon = document.querySelector('#weather_icon');
+const windSpeed = document.querySelector('#wind_speed');
+const cloudy = document.querySelector('#cloudy');
+const sunrise = document.querySelector('#sunrise');
+const sunset = document.querySelector('#sunset');
+
+// In openweather API temperature is indicated in Kelvin, convert it to Celcius
+function convertKelvinToCelcius(k) {
+    return Math.round(k - 272.15);
+}
+
+// Time is set as a timestamp, convert it to hours and minutes
+function convertTimestampToTime(stamp) {
+    const time = new Date(stamp * 1000);
+    return `${time.getHours()}:${time.getMinutes()}`;
+}
+
+// Use Google maps to show our location
+let map;
+function initMap() {
+    map = new google.maps.Map(document.getElementById('map'), {
+        center: { lat: 0, lng: 0 },
+        zoom: 10,
+    });
+}
+
+// Apply getWeather function to get weather by city name
+selectButton.addEventListener('click', () => {
+    if (cityInputField.value === '') {
+        alert('Select your city');
+    } else {
+        getWeather(`https://api.openweathermap.org/data/2.5/weather?q=${cityInputField.value}&appid=0b2171133666adf69c5fc412fc18bb00`);
+    }
+})
+
+function getWeather(url) {
+    fetch(url)
+        .then(response => response.json())
+        .then(data => {
+            // In case user misspelt the name of the city
+            if (data.cod === '404') {
```

Comment:
`.cod` => `.code`? It will still work, but only because this will always be false.